File: backend/ecommerce-site/src/main/java/com/mhr/ecommerce/dto/Purchase.java
package com.mhr.ecommerce.dto;

import com.mhr.ecommerce.entity.Address;
import com.mhr.ecommerce.entity.Customer;
import com.mhr.ecommerce.entity.Order;
import com.mhr.ecommerce.entity.OrderItem;
import lombok.Data;

import java.util.Set;

@Data
public class Purchase {

    private Customer customer;
    private Address address;
    private Order order;
    private Set<OrderItem> orderItems;

}

File: frontend/ecommerce-site/src/app/common/purchase.ts
import { Address } from './address';
import { Customer } from './customer';
import { Order } from './order';
import { OrderItem } from './order-item';

export class Purchase {
    customer: Customer;
    address: Address;
    order: Order;
    orderItems: OrderItem[];

    displayInfo(){
        console.log("Customer --> " 
        + this.customer.firstName + " ," + 
          this.customer.lastName + " ," + 
          this.customer.email + " ," +
          this.customer.password  +" ," +
          this.customer.role );
        // console.log("Address --> " + this.address.displayInfo());
        // console.log("Order --> " + this.order.displayInfo());
        // for (let i=0; i < this.orderItems.length; i++) {
        //     console.log("OrderItems --> " + this.orderItems[i].displayInfo());
        // }
    }

}
